_posts/2024-01-12-release-0_4_1.md

```diff
@@ -0,0 +1,35 @@
+---
+layout: post
+title:  "Kroxylicious release 0.4.1"
+date:   2023-01-12 00:00:00 +0000
+author: "Sam Barker"
+author_url: "https://www.github.com/SamBarker"
+categories:  releases kroxylcious topic-encryption
+---
+
+The Kroxylicious project is very pleased to announce the release of [Kroxylicious 0.4.1](https://github.com/kroxylicious/kroxylicious/releases/tag/v0.4.1). This release is focused on [fixing](https://github.com/kroxylicious/kroxylicious/issues/841) an issue where the proxy failed to preserve offsets while decrypting records. While we feel a little sheepish that our own testing (which we have improved at both [unit](https://github.com/kroxylicious/kroxylicious/pull/843) and [integration](https://github.com/kroxylicious/kroxylicious/pull/839/files#diff-f6ca390d21c7e5bcd3134c97c204d8394ada9217ae791da9da1c7255c8a97a75) levels) for 0.4.0 missed it we got a great bug report from [@giacomoa](https://github.com/giacomoa). Speaking of great bug reports this release also includes our first external contribution [PR#849](https://github.com/kroxylicious/kroxylicious/pull/849) from [@luozhenyu](https://github.com/luozhenyu) which fixes an issue in the SASL authentication handling.
+
+This release also includes some other small changes cleaning up examples and updating dependencies. 
+
+The whole Kroxylicious team is very excited to see people testing out the proxy and is really looking forward to hearing more about how & where people want to use it.
+
+### Encryption at Rest
+
+The `EnvelopeEncryption` filter works by intercepting all produce requests from applications and encrypts the Kafka records as they pass through Kroxylicious. On the consume path, the reverse happens - the filter intercepts the fetch responses and decrypts the records before they are sent to the application.
+We have more work planned to iterate on the implementation to reach production quality. Check out our [0.5.0 milestone](https://github.com/kroxylicious/kroxylicious/milestones/0.5.0) for details.
+
+The Kafka Cluster *never* sees the plain text of your records. Filter configuration is used to specify which topic(s) should be encrypted by which key.
```

Comment:
Shame we don't have documentation for this yet (https://github.com/kroxylicious/kroxylicious/issues/787), but nothing we can do here, I guess.